### PR TITLE
feat(server): add `setChannel` function

### PR DESCRIPTION
**Pull request by SeqviriouM**
Set received channels in state.
Tests included.

Related: #8

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
   "description": "chat",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "mocha --compilers js:babel/register --require ./server/test/test_helper.js  --recursive",
```
Comment (dimaip) [inline package.json]: It's better to call it `test-server`, to separate from client side tests.

File: server/test/test_core.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+            const state = Map();
+            const channels = List.of(Map({id: 1, name: '1', userIds: List.of(1,2)}));
+            console.log(channels);
+            const nextState = setChannels(state, channels);
+            expect(nextState).to.equal(Map({
+                channels: List.of(Map({id: 1, name: '1', userIds: List.of(1,2)}))
+            }));
+        });
+    });
+});
```
Comment (dimaip) [inline server/test/test_core.js]: See http://stackoverflow.com/questions/729692/why-should-files-end-with-a-newline

**Comment by dimaip** [comment]
:-1: see comments.

File: server/src/core.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+export function setChannels (state, channels) {
+  return state.set('channels', channels);
+}
+
+export function addUserToChannel (state, channelId, userId) {
+  let index = state.get('channels').map((item) => { return item.get('id'); }).indexOf(channelId);
+
+  return state.updateIn(['channels', index, 'userIds'], 0, userIds => userIds.push(userId) );
+}
```
Comment (Nitive) [inline server/src/core.js]: No line in the end of file

File: server/test/test_helper.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import chai from 'chai';
+import chaiImmutable from 'chai-immutable';
+
+chai.use(chaiImmutable);
```
Comment (Nitive) [inline server/test/test_helper.js]: No line in the end of file

File: server/src/core.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import {List, Map} from 'immutable';
+
+export function setChannels (state, channels) {
+  return state.set('channels', channels);
+}
+
+export function addUserToChannel (state, channelId, userId) {
+  let index = state.get('channels').map((item) => { return item.get('id'); }).indexOf(channelId);
```
Comment (Nitive) [inline server/src/core.js]: It would be const

**Comment by Nitive** [comment]
++

**Pull request closed without merging**